ouroboros-consensus/src/unstable-consensus-testlib/Test/Util/TestBlock.hs

```diff
@@ -550,7 +550,7 @@ data TestBlockLedgerConfig = TestBlockLedgerConfig {
   tblcHardForkParams :: HardFork.EraParams,
   -- | `Nothing` means an infinite forecast range.
   -- Instead of SlotNo, it should be something like "SlotRange"
-  tblcForecastRange  :: Maybe SlotNo
+  tblcForecastRange  :: !(Maybe SlotNo) -- StrictMaybe?
```

Comment:
Using `StrictMaybe` sounds good, we already use it in a few places.

Reply:
FTR: Another option would be to derive `NoThunks` via [`AllowThunk`](https://hackage.haskell.org/package/nothunks-0.2.1.0/docs/NoThunks-Class.html#t:AllowThunk), as per https://github.com/IntersectMBO/ouroboros-consensus/pull/1159#pullrequestreview-2156428591

Especially as we never "update" `LedgerConfig`s.